### DISCUSSION TITLE
MultiGPU support

### DISCUSSION THREAD

**Discussion by crystian** [crystian/ComfyUI-Crystools#20] [Show and tell]
I'm working on multi-gpu support, if you want to try it, please make this: 

```
cd custom_nodes\ComfyUI-Crystools
git checkout -b dev origin/gpu
```
Try it; your feedback and test are welcome! :)

**Comment by crystian** [comment]
In this branch, you will see on your consoles a lot of data for debugging for me, it's important; when I do a release, I will remove it.

If you switch a GPU/VRAM you will see on your server console the change and stoping/start the monitoring based on the id of the card

**Comment by GianPieroA** [comment]
I did clean local storage; can't find any user/default/comfy.settings.json (I am on windows). I did try to investigate also common used folders for settings (users/myuser, users/default, AppData/*) but there is no track of such a file. 

I then started ComfyUI with the latest gpu branch. At the beginning the situation is coherent: flags on settings reflect what the main panel is showing. Unfortunately unchecking a flag for the first time brings the widget value to -1 (see picture for VRAM0) 
![menu2](https://github.com/crystian/ComfyUI-Crystools/assets/55051802/e89e7dcb-8fec-4ebb-8e04-0d28b3c0bcc0)
Then checking it again makesthe relative widget disappear and there is where it starts to work on the contrary (checking it disables the widget and unchecking it re-enables the widget)
![menu3](https://github.com/crystian/ComfyUI-Crystools/assets/55051802/9b7a1698-3967-4ceb-99ae-28975e22b78e)

**Reply by crystian** [reply]
@GianPieroA  umm, did you update the branch? because I change the names on settings and I don't see the changes on the screenshot

```
git pull
```

**Reply by crystian** [reply]
the version should be 1.10.1-gpu (see on the console) when starts the server
![image](https://github.com/crystian/ComfyUI-Crystools/assets/3886806/10d66ca5-00cb-498b-b43c-bf5fc4e67014)

**Reply by GianPieroA** [reply]
@crystian  Sorry, it was my fault; I was not on the right brach. Now the version is aligned 1.10.1-gpu. Unfortunately the behaviour is still the same, also after cleaning localstorage and restarting comfyui:
- when it starts everything looks coherent: every widget is displayed and the relative checkbox in settings is set
- when I first interact with setting, unchecking a checkbox, the widget displays -1
- If, at this point, I re-check the checkbox then the widget disappears
- If I uncheck it, the widget reappears but with -1 . See picture where I just unchecked/re-checked/unchecked the CPU setting
![menu4](https://github.com/crystian/ComfyUI-Crystools/assets/55051802/443a2570-4fd3-4bae-a472-85db59fc9cb0)

**Reply by crystian** [reply]
ok, I'm working on that. It's so strange, but your rate is more than 0, right?

**Reply by crystian** [reply]
I need something that I believe is the problem, which version of comfyui do you have?, please 🙏 🙏  fill this **before** to update:

**Versions:**  
Copy the output of the console (4 parts), like this:
```
** Python version: 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
```
```
Total VRAM 12287 MB, total RAM 130996 MB
Set vram state to: NORMAL_VRAM
```
```
[Crystools INFO] Crystools version: 1.9.2
[Crystools INFO] CPU: AMD Ryzen 9 5950X 16-Core Processor - Arch: AMD64 - OS: Windows 10
[Crystools INFO] GPU/s:
[Crystools INFO] 0) NVIDIA GeForce RTX 3080 Ti
[Crystools INFO] NVIDIA Driver: 546.33
```
```
### Loading: ComfyUI-Manager (V1.25.3)
### ComfyUI Revision: 1886 [6a10640f] | Released on '2024-01-08'
```

after copied these information please update all, and ensure you use 1.10.1-gpu version 🙏 , and try again!

**Reply by crystian** [reply]
What order do you prefer? 
gpu0 > vram0 > gpu1 > vram1

![image](https://github.com/crystian/ComfyUI-Crystools/assets/3886806/f5544bc5-92d1-4e9e-9dc0-9f1f3ce09107)

gpu0 > gpu1 > vram0 > vram1

![image](https://github.com/crystian/ComfyUI-Crystools/assets/3886806/031754dd-2244-4cec-9a36-64bb47b34f05)

**Reply by crystian** [reply]
the version should be 1.10.2-gpu

**Reply by GianPieroA** [reply]
I think the second version  (gpu0 > gpu1 > vram0 > vram1) is less 'eyes distracting

**Comment by GianPieroA** [comment]
@crystian I will have a check tonight, today I am OOO

**Reply by GianPieroA** [reply]
```
** Python version: 3.11.6 (tags/v3.11.6:8b6ee5b, Oct  2 2023, 14:57:12) [MSC v.1935 64 bit (AMD64)]
```

```
Total VRAM 24576 MB, total RAM 65281 MB
Set vram state to: NORMAL_VRAM
```

```
[Crystools INFO] Crystools version: 1.10.1-gpu
[Crystools INFO] CPU: Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz - Arch: AMD64 - OS: Windows 10
[Crystools INFO] GPU/s:
[Crystools INFO] 0) NVIDIA GeForce GTX 1650 Ti
[Crystools INFO] GPU/s:
[Crystools INFO] 1) NVIDIA GeForce RTX 3090
```

```
### Loading: ComfyUI-Manager (V1.25)
### ComfyUI Revision: 1878 [6d281b4f] | Released on '2024-01-04'
```

Note: I didn't updated ComfyUI-Manager because of this debugging session. It may be the problem.

**Reply by GianPieroA** [reply]
ok, now my situation is:

```
[Crystools INFO] Crystools version: 1.10.2-gpu
[Crystools INFO] CPU: Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz - Arch: AMD64 - OS: Windows 10
[Crystools INFO] GPU/s:
[Crystools INFO] 0) NVIDIA GeForce GTX 1650 Ti
[Crystools INFO] 1) NVIDIA GeForce RTX 3090
[Crystools INFO] NVIDIA Driver: 546.33
```
```
### Loading: ComfyUI-Manager (V2.1)
### ComfyUI Revision: 1915 [ee2c5fa7] | Released on '2024-01-16'
```
And everythings works fine; i restarted ComfyUI a couple of times and did try a couple of workflows changing settings. Great job @crystian !

**Reply by crystian** [reply]
Nice! Thanks for your feedback; I will release it! :)
thanks again!

**Reply by GianPieroA** [reply]
@crystian Thanks to you!

**Comment by crystian** [comment]
release: 1.10.0
